### ISSUE TITLE
Fix S3971 FP: Support IAsyncDisposable

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#3639]
### Description

Rule [S3971](https://rules.sonarsource.com/csharp/RSPEC-3971) raises a false positive when `GC.SuppressFinalize` is called from [IAsyncDisposable.DisposeAsync](https://docs.microsoft.com/en-us/dotnet/api/system.iasyncdisposable.disposeasync?view=dotnet-plat-ext-3.1#System_IAsyncDisposable_DisposeAsync). This contradicts the best practices described in the API doc https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync#the-disposeasync-method

Source: https://community.sonarsource.com/t/false-positive-do-not-call-suppressfinalize/31740

### Repro steps

```
public class Custom : IAsyncDisposable
{
    public async ValueTask DisposeAsync()
    {
        await DisposeAsyncCore();

        GC.SuppressFinalize(this); // Noncompliant - FP, see: https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync#the-disposeasync-method
    }

    protected virtual ValueTask DisposeAsyncCore() => default;
}
```

### Expected behavior

S3971 should not raise an issue for this case.

### Actual behavior

False positive is reported.

### Known workarounds

Implement `IDisposable` instead.

### Related information

* C#/VB.NET Plugins Version: 8.13.1.21947

**Comment by dnnr** [comment]
Is it possible that there's a regression? I'm on v2025.3.1, and I'm getting S3971 complaints for calling SuppressFinalize in DisposeAsync.

**Comment by denis-troller** [comment]
> Is it possible that there's a regression? I'm on v2025.3.1, and I'm getting [S3971](https://sonarsource.github.io/rspec/#/rspec/S3971/csharp) complaints for calling SuppressFinalize in DisposeAsync.

Would you mind sharing a reproducer?

**Comment by pavel-mikula-sonarsource** [comment]
Hi, please open a new issue with the reproducer. Feel free to mention is issue as a related one / possible regression.

**Comment by dnnr** [comment]
Thanks, new issue opened. Just for the record: It doesn't look like a regression but it happens when using an intermediate interface.
